### PR TITLE
Pull request per issue #1011

### PR DESCRIPTION
**Pull request by lewisl**
Floating point values work.  Able to use symbols as keys in Dict, added test for same.  Possible error in runtest.jl for the way I include and use Pkg.ext.TOML.

Simplified: removed get methods. remove some methods. remove none methods.

All functional tests pass, modulo the error I may have introduced.

Please review code and process. Feedback needed as this is my first pull request to Julia.

Thanks

### CODE REVIEW

**Comment by KristofferC** [comment]
bors try

**Comment by KristofferC** [comment]
Thanks, seems like some tests failed:

https://travis-ci.org/JuliaLang/Pkg.jl/jobs/564222221

**Comment by lewisl** [comment]
Any diagnostics on what failed?  The changes were all confined to the Pkg/ext directory.

**Comment by KristofferC** [comment]
Yes, we run the tests for the TOML package as well:

```
Lookups: Error During Test at /home/travis/build/JuliaLang/Pkg.jl/ext/TOML/test/runtests.jl:135
  Got exception outside of a @test
  MethodError: no method matching get(::Array{String,1})
  Closest candidates are:
    get(::AbstractArray, !Matched::Integer, !Matched::Any) at abstractarray.jl:1146
    get(::AbstractArray, !Matched::Tuple{}, !Matched::Any) at abstractarray.jl:1147
    get(::AbstractArray, !Matched::Tuple{Vararg{Int64,N}} where N, !Matched::Any) at abstractarray.jl:1148
    ...
 ```

etc

**Comment by lewisl** [comment]
This looks like a different set of tests.  Runtests.jl in Pkg/ext/TOML/test runs 244 tests, all of which pass.  This shows 135 tests.

Some that failed refer to TOML.get.  There shouldn’t be any tests for that method because I removed it.

Can there be a name conflict between Pkg/ext/TOML/src/TOML.jl and the TOML that is still in the main package registry?

Is there a way I can see what tests are being run?

**Comment by lewisl** [comment]
Not sure we are running the same tests.  Orginally, TOML provided a method for get that did require a key—I got rid of that because it essentially just embedded the @assert !isnull, which wasn’t needed.  But, when I run the tests locally in my fork, they all pass.  I might have some name space mangling going on.

Tomorrow or Sunday, I’ll run what I have locally and send you the output to see.  Then, perhaps you can point me to how to clean up the pull request with the right tests and dependencies.  I was afraid my efforts might be more trouble to you than they are worth.  Just try to push it back onto me.  Once I am sorted out, I’ll be more useful.

**Comment by lewisl** [comment]
Here are the results of the tests as I run them:

```
julia> include("runtests.jl")
Test Summary: | Pass  Total
TOML          |  244    244
```

But, you can see that I am running them in the repl, which is undoubtedly wrong.

Within runtests.jl, I access the TOML module as, which works for my local copy of the git repo:

```
include("../src/TOML.jl")

using .TOML  
import TOML: linecol, whitespace, comment, newline, expect, lookup, Parser, parse
```

This is also probably the wrong way to do it.

Finally, I added this as the last test in the "Printer" test group (end of runtest.jl).  See if this is what you see in the pull request:

```
    # symbols as Keys
    io = IOBuffer()
    d = Dict(:products=>[Dict("name"=>"me","cnt"=>3),Dict{String,Any}(),Dict(:name=>"you","cnt"=>5)])
    TOML.print(io, d)
    @test String(take!(io)) == "[[products]]\nname = \"me\"\ncnt = 3\n[[products]]\n[[products]]\ncnt = 5\nname = \"you\"\n"
```

When I go to github and look at the last commit of my pull request, 780fb9, the test above is added in the diff.  I wonder if you ran the tests for the overall Pkg package that test its use of TOML?  Of, course, that should work, too.

So, I screwed up the workflow.

**Comment by lewisl** [comment]
Naturally, I don’t have access to Travis CI, but I looked at the log and examined several of the test cases that raised exceptions.  I could not find those tests in the runtests.jl that are part of Pkg/ext/TOML.

I could not find several of those tests (that raised exceptions) in the runtests.jl that is part of @wildart’s latest version of TOML.

Maybe we could clarify what testfile is being run.  Then, I could run it.

I note that the travis.yml in Pkg/ext/TOML refers to Julia 0.6. Seems like perhaps stuff in the github repo doesn’t match Travis, though Travis should read from github (indeed, the commit hash is correct).

**Comment by lewisl** [comment]
Last comment for a while.   Below I have pasted in the runtests.jl which exists in my pull request at Pkg/ext/TOML/test/runtest.jl.  It seems pretty clear that this is NOT the test file that Travis is running in the build test.  But, this is barely changed from the test file in the repo when I forked it.  What test file is Travis using?  I wonder if it is the TOML test file in the old Base registry, which would be quite different from both @wildart's and the tests in the pull request.

In particular, I removed tests using TOML.get(parse(p)) as the TOML packages method on get has been removed--in the diff of the test file there are only 2 such tests.  I guess I don't know how to test the new code except with tests that match the new code.

I don't have access to Travis (I shouldn't) so I have to run the test file locally.

Here it is:
```
# This file is a part of Julia. License is MIT: https://julialang.org/license

include("../src/TOML.jl")

using .TOML  
import TOML: linecol, whitespace, comment, newline, expect, lookup, Parser, parse


using Test
using Dates

macro testval(s, v)
    f = "foo = $s"
    :( @test TOML.get(parse(Parser($f)))["foo"] == $v )
end

macro fail(s...)
    teststr = s[1]
    errstr = ""
    debug = false
    if length(s) == 2
        if isa(s[2], String)
            errstr = s[2]
        elseif isa(s[2], Expr)
            debug = s[2].args[1] == :debug
        end
    end
    if length(s) == 3
        errstr = s[2]
        debug = s[3].args[1] == :debug
    end

    # macro vars
    pvar = :pv
    ppvar = :ppv

    # debugging report
    dbgexp = if debug
        quote
            println("\nTEST FAIL: ", escape_string($teststr))
            if length($pvar.errors) > 0
                println("ERRORS:")
                for e in $pvar.errors
                    println(e)
                end
            end
            if !TOML.isnull($ppvar)
                println("RESULT:")
                println(TOML.get($ppvar))
            end
        end
    else
        :()
    end

    # error for comparison
    errtsts = if !isempty(errstr)
        quote
            @test length($pvar.errors)>0
            @test findlast(m->m==$errstr, map(e->e.msg, $pvar.errors)) != 0
        end
    else
        :()
    end

    quote
        local $pvar = Parser($teststr)
        local $ppvar = parse($pvar)
        $dbgexp
        $errtsts
        @test TOML.isnull($ppvar)
    end
end

macro success(s...)
    teststr = s[1]
    debug = false
    if length(s) == 2 && isa(s[2], Expr)
        debug = s[2].args[1] == :debug
    end

    # macro vars
    pvar = :pv
    ppvar = :ppv

    # debugging report
    dbgexp = if debug
        quote
            println("\nTEST SUCCESS: ", escape_string($teststr))
            if length($pvar.errors) > 0
                println("ERRORS:")
                for e in $pvar.errors
                    println(e)
                end
            end
            if !TOML.isnull($ppvar)
                println("RESULT:")
                println(TOML.get($ppvar))
            end
        end
    else
        :()
    end

    quote
        local $pvar = Parser($teststr)
        local $ppvar = parse($pvar)
        $dbgexp
        @test !TOML.isnull($ppvar)
    end
end

@testset "TOML" begin
@testset "Parser" begin

    @testset "Parser internal functions" begin
        test = """
   #[test]\r
   [test]
   foo = "bar"
"""
        p = Parser(test)
        @test whitespace(p)
        @test position(p) == 4
        @test comment(p)
        @test position(p) == 13
        @test whitespace(p)
        @test position(p) == 16
        @test !expect(p, 'a')
        @test expect(p, '[')
        @test position(p) == 17
    end


    @testset "Lookups" begin
        p = Parser("""hello."world\\t".a.0.'escaped'.value""")
        @testset for (p, s) in zip(TOML.get(lookup(p)), ["hello"; "world\t"; "a"; "0"; "escaped"; "value"])
            @test p == s
        end

        p = Parser("")
        @test TOML.get(lookup(p)) == String[]

        p = Parser("value")
        @test TOML.get(lookup(p)) == String["value"]

        p = Parser("\"\"")
        #TODO: @test get(lookup(p)) == String[""]

    end

    @testset "New line" begin
        @success("""
[project.A]\r
name = \"splay2\"\r
\r
[project]\r
\r
name = \"splay\"\r
version = \"0.1.0\"\r
authors = [\"alex@crichton.co\"]\r
\r
[[lib]]\r
\r
path = \"lib.rs\"\r
name = \"splay\"\r
description = \"\"\"
A Rust implementation of a TAR file reader and writer. This library does not\r
currently handle compression, but it is abstract over all I/O readers and\r
writers. Additionally, great lengths are taken to ensure that the entire\r
contents are never required to be entirely resident in memory all at once.\r
\"\"\"
[[lib.aaa]]\r
bbb = \"aaa\"\r
""")

        @fail("\r")

        @fail("a = [ \r ]")

        @fail("a = \"\"\"\r\"\"\"")

        @fail("a = \"\"\"\\  \r  \"\"\"")

        @success("a = '''\r'''")

        @success("a = '\r'")

        @fail("a = \"\n\"")

        @fail("a = '\n'")

        p = Parser("
            foo = \"\"\"\\\r\n\"\"\"
            bar = \"\"\"\\\r\n   \r\n   \r\n   a\"\"\"
        ")
        res = TOML.get(parse(p))
        @test res["foo"] == ""
        @test res["bar"] == "a"

        @fail("0=0r=false", "expected a newline after a key")

        @fail("""
0=""o=""m=""r=""00="0"q=\"\"\"0\"\"\"e=\"\"\"0\"\"\"
""", "expected a newline after a key")

        @fail("""
[[0000l0]]
0="0"[[0000l0]]
0="0"[[0000l0]]
0="0"l="0"
""", "expected a newline after a key")

        @fail("""
0=[0]00=[0,0,0]t=["0","0","0"]s=[1000-00-00T00:00:00Z,2000-00-00T00:00:00Z]
""", "expected a newline after a key")

        @fail("""
0=0r0=0r=false
""", "expected a newline after a key")

        @fail("""
0=0r0=0r=falsefal=false
""", "expected a newline after a key")

    end

    @testset "Offset->(Line, Col) " begin
        p = Parser("ab\ncde\nf")
        @test linecol(p, 0) == (1,0)
        @test linecol(p, 1) == (1,1)
        @test linecol(p, 4) == (2,1)
        @test linecol(p, 5) == (2,2)
        @test linecol(p, 8) == (3,1)
        @test linecol(p, 9) == (3,0)
    end

    @testset "Strings" begin
        p = Parser("""
bar = "\\U00000000"
key1 = "One\\nTwo"
key2 = \"\"\"One\nTwo\"\"\"
key3 = \"\"\"
One
Two\"\"\"

key4 = "The quick brown fox jumps over the lazy dog."
key5 = \"\"\"
The quick brown \\


  fox jumps over \\
    the lazy dog.\"\"\"
key6 = \"\"\"\\
       The quick brown \\
       fox jumps over \\
       the lazy dog.\\
       \"\"\"

# What you see is what you get.
winpath  = 'C:\\Users\\nodejs\\templates'
winpath2 = '\\\\ServerX\\admin\$\\system32\\'
quoted   = 'Tom "Dubs" Preston-Werner'
regex    = '<\\i\\c*\\s*>'
regex2 = '''I [dw]on't need \\d{2} apples'''
lines  = '''
The first newline is
trimmed in raw strings.
    All other whitespace
    is preserved.
'''
""")
        res = TOML.get(parse(p))

        @test res["bar"]  == "\0"
        @test res["key1"] == "One\nTwo"
        @test res["key2"] == "One\nTwo"
        @test res["key3"] == "One\nTwo"

        msg = "The quick brown fox jumps over the lazy dog."
        @test res["key4"] == msg
        @test res["key5"] == msg
        @test res["key6"] == msg

        @test res["winpath"] == "C:\\Users\\nodejs\\templates"
        @test res["winpath2"] == "\\\\ServerX\\admin\$\\system32\\"
        @test res["quoted"] == """Tom "Dubs" Preston-Werner"""
        @test res["regex"] == """<\\i\\c*\\s*>"""
        @test res["regex2"] == """I [dw]on't need \\d{2} apples"""

        @test res["lines"] == "The first newline is\ntrimmed in raw strings.\n    All other whitespace\n    is preserved.\n"


        @testval("''", "")
        @testval("\"\"", "")
        @testval("\"\"\"\n\n\n\"\"\"", "\n\n")

        #@fail("foo = \"\\uD800\"")

        @fail("foo = \"\\uxx\"")
        @fail("foo = \"\\u\"")
        @fail("foo = \"\\")
        @fail("foo = '")

    end

    @testset "Numbers" begin
        @fail("a = 00")
        @fail("a = -00")
        @fail("a = +00")
        @fail("a = 00.0")
        @fail("a = -00.0")
        @fail("a = +00.0")
        @fail("a = 9223372036854775808")
        @fail("a = -9223372036854775809")

        @fail("a = 0.")
        @fail("a = 0.e")
        @fail("a = 0.E")
        @fail("a = 0.0E")
        @fail("a = 0.0e")
        @fail("a = 0.0e-")
        @fail("a = 0.0e+")
        @fail("a = 0.0e+00")

        @testval("1.0", 1.0)
        @testval("1.0e0", 1.0)
        @testval("1.0e+0", 1.0)
        @testval("1.0e-0", 1.0)
        @testval("1.001e-0", 1.001)
        @testval("2e10", 2e10)
        @testval("2e+10", 2e10)
        @testval("2e-10", 2e-10)
        @testval("2_0.0", 20.0)
        @testval("2_0.0_0e0_0", 20.0)
        @testval("2_0.1_0e1_0", 20.1e10)

        @fail("4")

        @testval("1_0", 10)
        @testval("1_0_0", 100)
        @testval("1_000", 1000)
        @testval("+1_000", 1000)
        @testval("-1_000", -1000)

        @fail("foo = 0_")
        @fail("foo = 0__0")
        @fail("foo = __0")
        @fail("foo = 1_0_")

    end

    @testset "Booleans" begin

        @testval("true", true)
        @testval("false", false)

        @fail("foo = true2")
        @fail("foo = false2")
        @fail("foo = t1")
        @fail("foo = f1")

    end

    @testset "Datetime" begin

        @testval("2016-09-09T09:09:09Z", DateTime(2016,9,9,9,9,9))
        @testval("2016-09-09T09:09:09.0Z", DateTime(2016,9,9,9,9,9))
        @testval("2016-09-09T09:09:09.0+10:00", DateTime(2016,9,9,19,9,9))
        @testval("2016-09-09T09:09:09.012-02:00", DateTime(2016,9,9,7,9,9,12))

        @fail("foo = 2016-09-09T09:09:09.Z", "malformed date literal")
        @fail("foo = 2016-9-09T09:09:09Z", "malformed date literal")
        @fail("foo = 2016-09-09T09:09:09+2:00", "malformed date literal")
        @fail("foo = 2016-09-09T09:09:09-2:00", "malformed date literal")
        @fail("foo = 2016-09-09T09:09:09Z-2:00", "expected a newline after a key")

    end


    @testset "Keys" begin

        p = Parser("
            foo = 3
            foo_3 = 3
            foo_-2--3--r23f--4-f2-4 = 3
            _ = 3
            - = 3
            8 = 8
            \"a\" = 3
            \"!\" = 3
            \"a^b\" = 3
            \"\\\\\\\"\" = 3
            \"character encoding\" = \"value\"
            'ʎǝʞ' = \"value\"
        ")
        res = TOML.get(parse(p))

        @test haskey(res, "foo")
        @test haskey(res, "-")
        @test haskey(res, "_")
        @test haskey(res, "8")
        @test haskey(res, "foo_3")
        @test haskey(res, "foo_-2--3--r23f--4-f2-4")
        @test haskey(res, "a")
        @test haskey(res, "!")
        @test haskey(res, "\\\"")
        @test haskey(res, "character encoding")
        @test haskey(res, "ʎǝʞ")

        @fail("key\n=3")
        @fail("key=\n3")
        @fail("key|=3")
        @fail("\"\"=3")
        @fail("=3")
        @fail("\"\"|=3")
        @fail("\"\n\"|=3")
        @fail("\"\r\"|=3")

    end


    @testset "Tables" begin

        @fail("[]")
        @fail("[.]")
        @fail("[\"\".\"\"]")
        @fail("[a.]")
        @fail("[\"\"]")
        @fail("[!]")
        @fail("[\"\n\"]")
        @fail("[a.b]\n[a.\"b\"]")
        @fail("[']")
        @fail("[''']")
        @fail("['''''']")
        @fail("['\n']")
        @fail("['\r\n']")

        p = Parser("
            [a.\"b\"]
            [\"f f\"]
            [\"f.f\"]
            [\"\\\\\\\"\"]
            ['a.a']
            ['\"\"']
        ")
        res = TOML.get(parse(p))
        @test haskey(res, "a.a")
        @test haskey(res, "f f")
        @test haskey(res, "f.f")
        @test haskey(res, "\\\"")
        @test haskey(res, "\"\"")
        @test haskey(res["a"], "b")


        @test haskey(TOML.get(parse(Parser("[foo]"))), "foo")


        @testset "Inline Tables" begin

            @success("a = {}")
            @success("a = {b=1}")
            @success("a = {   b   =   1    }")
            @success("a = {a=1,b=2}")
            @success("a = {a=1,b=2,c={}}")
            @fail("a = {a=1,}")
            @fail("a = {,}")
            @fail("a = {a=1,a=1}")
            @fail("a = {\n}")
            @fail("a = {")
            @success("a = {a=[\n]}")
            @success("a = {\"a\"=[\n]}")
            @success("a = [\n{},\n{},\n]")

        end


        @testset "Redefinition" begin
            @fail("
                [a]
                foo=\"bar\"
                [a.b]
                foo=\"bar\"
                [a]
            ", "redefinition of table `a`")
            @fail("
                [a]
                foo=\"bar\"
                b = { foo = \"bar\" }
                [a]
            ", "redefinition of table `a`")
            @success("
                [a.b]
                c = {}
                [a]
                d = 1
            ")
            @fail("
                [a.b]
                c = 1
                [a]
                b = {}
            ", "duplicate key `b`")
            @fail("
                [a]
                b = {}
                [a]
            ", "redefinition of table `a`")
        end

        @testset "Nesting" begin
            @fail("
                a = [2]
                [[a]]
                b = 5
            ", "expected type `TOML.Table`, found type `$(Int)`")
            @fail("
                a = 1
                [a.b]
            ", "key `a` was not previously a table")
            @fail("
                a = []
                [a.b]
            ", "array `a` does not contain tables")
            @success("
                a = [{}]
                [a.b]
            ")
            @fail("
                a = []
                [[a.b]]
            ", "array `a` does not contain tables")
            @fail("
                [a]
                b = { c = 2, d = {} }
                [a.b]
                c = 2
            ", "duplicate key `c`")
        end
    end

    @testset "Arrays" begin

        p = Parser("""
[[foo]]
  #…
  [foo.bar]
    #…
[[foo]] # ...
  #…
  [foo.bar]
    #...
""")
        res = parse(p)
        @test haskey(res, "foo")
        arr = res["foo"]
        @test length(arr) == 2
        @test repr(typeof(arr[1])) == "TOML.Table"
        @test haskey(arr[1], "bar")
        @test haskey(arr[2], "bar")

        p = Parser("""
[[fruit]]
  name = "apple"
  [fruit.physical]
    color = "red"
    shape = "round"
  [[fruit.variety]]
    name = "red delicious"
  [[fruit.variety]]
    name = "granny smith"
[[fruit]]
  name = "banana"
  [[fruit.variety]]
    name = "plantain"
""")
        res = parse(p)
        @test haskey(res, "fruit")
        fruit = res["fruit"]

        @test length(fruit) == 2
        @test repr(typeof(fruit[1])) == "TOML.Table"
        apple = fruit[1]
        banana = fruit[2]

        @test haskey(apple, "name")
        @test apple["name"] == "apple"
        @test haskey(apple, "physical")
        @test apple["physical"]["color"] == "red"
        @test apple["physical"]["shape"] == "round"
        @test apple["variety"][1]["name"] == "red delicious"
        @test apple["variety"][2]["name"] == "granny smith"

        @test haskey(banana, "name")
        @test banana["name"] == "banana"
        @test banana["variety"][1]["name"] == "plantain"

        res = TOML.parse("""
[[products]]
name = "Hammer"
sku = 738594937

[[products]]

[[products]]
name = "Nail"
sku = 284758393
color = "gray"
""")
        @test haskey(res, "products")
        products = res["products"]
        @test isa(products, Array)
        @test length(products) == 3
        @test products[1]["name"] == "Hammer"
        @test products[1]["sku"] == 738594937
        @test length(products[2]) == 0
        @test products[3]["name"] == "Nail"
        @test products[3]["sku"] == 284758393
        @test products[3]["color"] == "gray"

        @fail("""array = [
        #         "This might most likely happen in multiline arrays",
        #         Like here,
        #         "or here,
        #         and here"
        #         ]     End of array comment, forgot the #
        """)
    end

end

@testset "Printer" begin
    res1 = TOML.parse("""
title = "TOML Example"
[owner]
bio = \"\"\"GitHub Cofounder & CEO
Likes tater tots and beer.\"\"\"
dob = 1979-05-27T07:32:00Z # First class dates? Why not?

[database]
server = "192.168.1.1"
ports = [ 8001, 8001, 8002 ]
enabled = true

[servers]

  # You can indent as you please. Tabs or spaces. TOML don't care.
  [servers.alpha]
  ip = "10.0.0.1"
  dc = "eqdc10"

  [servers.beta]
  ip = "10.0.0.2"
  dc = "eqdc10"
  country = "中国" # This should be parsed as UTF-8

[clients]
data = [ ["gamma", "delta"], [1, 2] ] # just an update to make sure parsers support it

  [[products]]
  name = "Hammer"
  sku = 738594937

  [[products]]
  name = "Nail"
  sku = 284758393
  color = "gray"
""")
    io = IOBuffer()
    TOML.print(io, res1)
    seekstart(io)
    res2 = TOML.parse(io)
    @test res1 == res2

    # sorting with non-Function by
    io = IOBuffer()
    d = Dict("1.10" => 10, "1.1" => 1)
    TOML.print(io, d; sorted=true, by = VersionNumber)
    @test String(take!(io)) == "\"1.1\" = 1\n\"1.10\" = 10\n"

    # symbols as Keys
    io = IOBuffer()
    d = Dict(:products=>[Dict("name"=>"me","cnt"=>3),Dict{String,Any}(),Dict(:name=>"you","cnt"=>5)])
    TOML.print(io, d)
    @test String(take!(io)) == "[[products]]\nname = \"me\"\ncnt = 3\n[[products]]\n[[products]]\ncnt = 5\nname = \"you\"\n"

end
end
```

**Comment by KristofferC** [comment]
You would run the test by

```
cd ext/TOML
julia --project
import Pkg; Pkg.test()
```

This shows the same errors as in Travis. Even in the code you pasted there are a bunch of `TOML.get` calls.

**Comment by lewisl** [comment]
Because there is no toml get it’s base.get. Anyway, thanks for the test file.

- Lewis

On Jul 26, 2019, at 9:07 PM, Kristoffer Carlsson <notifications@github.com<mailto:notifications@github.com>> wrote:


You would run the test by

cd ext/TOML
julia --project
import Pkg; Pkg.test()


This shows the same errors as in Travis. Even in the code you pasted there are a bunch of TOML.get calls.

**Comment by lewisl** [comment]
Just made new pull request.

Sorry for the past mess. Much progress.
Latest pull request should be much cleaner.

My local environment installed @wildart.TOML. When I ran tests, I included my fork of Pkg/ext/TOML, and the import imported @wildart. That is why I saw the tests all pass: I was testing a horrible blend of both modules. I rm'ed @wildart.TOML.
I removed TOML.get(... from the tests as I remove that method from the code.
I added the fork to my LOAD_PATH so I could drop the include. The name of the module is no longer mangled with a dot, which had caused tests on type TOML.Table to fail (I had hacked the tests). I set these back as the original file defined these tests. isa(foo, TOML.Table). These pass.
I added 4 tests that @wildart included which were not in Pkg/ext/TOML/test.
I run the tests as you prescribed with these results:
julia> import Pkg; Pkg.test()
   Testing TOML
 Resolving package versions...
Test Summary: | Pass  Total
TOML          |  252    252
   Testing TOML tests passed 
Note: there are now 252 tests. I could not locate the test file you also ran containing 130 or so tests. This must be from the old version of TOML carried around as part of Base??? I think this is accessible in Travis, but I could not find it (even though I enrolled in Travis).

I hope that Pkg/ext/TOML/test/runtests.jl should do much better in Travis now.

If you point me to the other test file I can either add it as a second test file (which might break the workflow) or--better--add those tests which are not duplicates. Note that the older tests will still reference the method TOML.get.

Note there were 2 ways to fix the floating point value bug: either import Base: get so that TOML could add a method. Or get rid of the TOML.get method and then the import is unnecessary. I did the latter.

**Comment by lewisl** [comment]
Here is the Travis build for my latest pull request on my fork of Pkg:
https://travis-ci.org/lewisl/Pkg.jl/builds

**Comment by KristofferC** [comment]
bors try

File: ext/TOML/src/parser.jl [hunk 1]
```diff
@@ -54,13 +55,7 @@ Base.position(p::Parser) = Int(position(p.input))+1
...
 Base.read(p::Parser) = p.currentchar = read(p.input, Char)
 
-NONE() = nothing
-NONE(::Type{T}) where {T} = nothing
-SOME(v::T) where {T} = v
-NONE(::Type{String}, p::Parser) = (take!(p.charbuffer); nothing)
-SOME(::Type{String}, p::Parser) = String(take!(p.charbuffer))
-isnull(x) = x === nothing
-get(x) = (@assert !isnull(x); x)
+isnull(x) = x === nothing  # works with older versions of Julia as isnothing() only introduced in 1.1
```
Comment (KristofferC) [inline ext/TOML/src/parser.jl]: We should change this to just use `x === nothing` inline since that currently works better with the compiler, but that can be done in another PR.

**Comment by KristofferC** [comment]
I tried to squash the commits on this branch but something weird happened. Reopened the PR at https://github.com/JuliaLang/Pkg.jl/pull/1272.
